### ISSUE TITLE
Jump executes after digging obstruction

### ISSUE THREAD

**Issue by S-S-X** [S-S-X/jumpdrive#1]
situation one
===========
1) build valid fleet with one fleet controller
2) execute luac script that does 'reset'
3) do via luac 'show' -> nothing as would be jump in itself
4) do via formspec set coords
5) do via luac 'show' -> markers
6) do via luac 'jump' -> obstructed bc jumping straight up and beacon is in the way
7) dig beacon -> jump is executed
@SwissalpS

**Comment by S-S-X** [comment]
I could not reproduce this issue at all, tried with different configuration and many times, added some logging and last jump command always says "obstructed" and cancels timer.

-

@SwissalpS **could you provide lua controller code used for this?**

-

# How I tested this:

### Top view, y at backbone / middle row:
```
GGXXXXXXX
G XXXXXXX
BJ####J##
SGGGXXXXX
XXXXXXXXX
```
### Side view, x at backbone / middle row:
```
   I    F
SJ####J##
PPPPPPPXX
```
### Components:
```
# = Backbone
J = Jumpdrive
F = Fleetcontroller
I = Beacon
G = solar Generator
B = Battery box
S = Switching
P = Power cable
X = Filler steel block
```
Fleet controller at 0,0,0
Radius 2 for both jumpdrives
Default digiline channel

## Actions in order:

1. digiline_send('fleetcontroller', { command = 'reset' })
2. digiline_send('fleetcontroller', { command = 'show' })
    logging complains: jump into itself
3. formspec, changed field "Y:" to "10" and clicked "Save"
3. digiline_send('fleetcontroller', { command = 'show' })
    logging complains: target is obstructed
4. digiline_send('fleetcontroller', { command = 'jump' })
    logging complains: target is obstructed.
5. dig beacon away
    Nothing happens

## After failure to reproduce issue:
Move beacon to another section and do exact same things, results:
4. First section jumps away
    logging complains about second section (one with fleet controller): target is obstructed

This behavior is same as #3 and is expected to happen, jump was executed even while fleet controller reported that jump is not safe.

**Comment by S-S-X** [comment]
```lua
if event.type == 'program' then
  print('STAGE: ' .. (mem.stage or 'init'))
  if mem.stage == 1 then
    digiline_send('fleetcontroller', { command = 'reset' })
  elseif mem.stage == 2 then
    digiline_send('fleetcontroller', { command = 'show' })
  elseif mem.stage == 3 then
    print(' !!!!!!!!!!!!!!!!!!!!!!! formspec, changed field "Y:" to "10" and clicked "Save"')
  elseif mem.stage == 4 then
    digiline_send('fleetcontroller', { command = 'show' })
  elseif mem.stage == 5 then
    digiline_send('fleetcontroller', { command = 'jump' })
  elseif mem.stage == 6 then
    print(' !!!!!!!!!!!!!!!!!!!!!!!!! dig beacon away')
  else
    mem.stage = 0
  end
  mem.stage = mem.stage + 1
end
```

**Comment by S-S-X** [comment]
## Guard against stacking async operations

async operations can be stacked and more started before earlier are finished. happens when commands are sent in tight loop.

reply with error message when user attempts to execute anything via digilines while another async cmd is running.

This should fix #1 and #4

**Comment by S-S-X** [comment]
https://github.com/S-S-X/jumpdrive/commit/1f0ab3e172e2ad3bd3c4ed9602648e2e0a5dace2
